engine/src/main/java/org/terasology/network/ClientInfoComponent.java

```diff
@@ -24,5 +24,14 @@
  */
 @Replicate
 public final class ClientInfoComponent implements Component {
-    // just a marker
+
+    /**
+     * When a client connects, the game searches a client info component for the client id ({@link Client#getId()}).
+     * If it finds one it is gets reused, otherwise a new one will be created.
+     *
+     * The field does not get replicated as there is no need to tell the clients the player ids.
```

Comment:
I was tempted to disagree with this, but then there is no situation where a player id would be useful where the entityref to the client info component wouldn't work too.